src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -165,7 +179,40 @@ async def trigger(self):
             )
 
             raw_results = self._raw_results_received.get(timeout=self.timeout_s)
-            LOGGER.info(f"Zocalo: found {len(raw_results['results'])} crystals.")
+            source_of_first_results, source_of_second_results = (
+                ("CPU", "GPU")
+                if not raw_results["ispyb_ids"].get("gpu")
+                else ("GPU", "CPU")
+            )
```

Comment:
Could: There's a bit of an inbuilt assumption here that there's not two from the same source, which is probably fine but I also don't really like the multiple use of strings and stuff. What about something like:
```python
class ZocaloSource(enum):
    CPU = "CPU"
    GPU = "GPU"

def source_from_results(results):
    return ZocaloSource.GPU if results["ispyb_ids"].get("gpu") else ZocaloSource.CPU
```
then each time we get results call `source_from_results(results)`

Reply:
I like the function, but I think it's useful to assume that they're not for the same source so that the logging message

```
LOGGER.warning(
                        f"Zocalo results from {source_of_second_results} timed out. Using results from {source_of_first_results}"
                    )
```
Is more useful.